### ISSUE TITLE
[java] False positive for NonStaticInitializer in anonymous class

### ISSUE THREAD

**Issue by oowekyala** [pmd/pmd#1853]
Rule: [NonStaticInitializer](https://pmd.github.io/pmd-6.15.0/pmd_rules_java_errorprone.html#nonstaticinitializer)

An instance initializer is the only way to initialize an anonymous class. The alternative is to create a local class with a constructor but quite honestly it's verbose and not that useful.

Test case:

```java
public class Foo {
       public Animation getStatusTransition() {
          return new Transition() {

            {
                setCycleDuration(Duration.millis(1200));
            }

            @Override
            protected void interpolate(double frac) {
               // magic
            }
        };
     }
}
```
